src/controllers/project.controller.ts

```diff
@@ -746,6 +738,47 @@ export class ProjectController {
 		}
 	}
 
+	async setProjectResourcesDirectory(resourcesPath: string): Promise<void> {
+		try {
+			const newDirectoryUri = Uri.file(resourcesPath);
```

Comment:
`currentDirectoryUri`

Reply:
Check if `Uri.file(resourcesPath);` and other functions can throw exceptions.
Insert to `try` only functions or actions that can fail.